controller/templates/controller/partials/state_machine.html

```diff
@@ -1,6 +1,42 @@
 {% load render_table from django_tables2 %}
 <div _="on load if #argsDialog is not null remove #argsDialog"></div>
-<form>
-  {% csrf_token %}
-  {% render_table table %}
-</form>
+<div class="row">
+  <div class="col-md-auto">
+    <form>
+      {% csrf_token %}
+      {% render_table table %}
+    </form>
+  </div>
+  <div class="col-md-auto">
+    <pre class="mermaid">
+      flowchart TD
+
+      %% FSM states (nodes).
+      classDef default fill:#b5b3ae,stroke:black,stroke-width:2px
+
+      %% FSM transitions (edges).
+      linkStyle default background-color:#b5b3ae,stroke-width:2px
+
+      {{ flowchart }}
+    </pre>
+  </div>
+</div>
+<script>
+  // Initialize mermaid diagrams.
+  mermaid.initialize({
```

Comment:
I guess there's no way of doing this without JS, right?

Reply:
Afraid not.